spec/operators/combineAll-spec.ts

```diff
@@ -1,453 +1,522 @@
-    expectSubscriptions(e2.subscriptions).toBe(e2subs);
+    testScheduler.run(({ cold, expectObservable, expectSubscriptions }) => {
+      const e1 = cold(' -');
+      const e1subs = '  ^';
+      const e2 = cold(' -');
+      const e2subs = '  ^';
+      const expected = '-';
+
+      const result = of(e1, e2).pipe(combineAll((x, y) => x + y));
+
+      expectObservable(result).toBe(expected);
+      expectSubscriptions(e1.subscriptions).toBe(e1subs);
+      expectSubscriptions(e2.subscriptions).toBe(e2subs);
+    });
   });
 
   it('should work with never and empty', () => {
-    const e1 = cold( '-');
-    const e1subs =   '^';
-    const e2 = cold( '|');
-    const e2subs =   '(^!)';
-    const expected = '-';
-
-    const result = of(e1, e2).pipe(combineAll((x, y) => x + y));
-
-    expectObservable(result).toBe(expected);
-    expectSubscriptions(e1.subscriptions).toBe(e1subs);
-    expectSubscriptions(e2.subscriptions).toBe(e2subs);
+    testScheduler.run(({ cold, expectObservable, expectSubscriptions }) => {
+      const e1 = cold(' -');
+      const e1subs = '  ^';
+      const e2 = cold(' |');
+      const e2subs = '  (^!)';
+      const expected = '-';
+
+      const result = of(e1, e2).pipe(combineAll((x, y) => x + y));
+
+      expectObservable(result).toBe(expected);
+      expectSubscriptions(e1.subscriptions).toBe(e1subs);
+      expectSubscriptions(e2.subscriptions).toBe(e2subs);
+    });
   });
 
   it('should work with empty and never', () => {
-    const e1 = cold( '|');
-    const e1subs =   '(^!)';
-    const e2 = cold( '-');
-    const e2subs =   '^';
-    const expected = '-';
-
-    const result = of(e1, e2).pipe(combineAll((x, y) => x + y));
-
-    expectObservable(result).toBe(expected);
-    expectSubscriptions(e1.subscriptions).toBe(e1subs);
-    expectSubscriptions(e2.subscriptions).toBe(e2subs);
+    testScheduler.run(({ cold, expectObservable, expectSubscriptions }) => {
+      const e1 = cold(' |');
+      const e1subs = '  (^!)';
+      const e2 = cold(' -');
+      const e2subs = '  ^';
+      const expected = '-';
+
+      const result = of(e1, e2).pipe(combineAll((x, y) => x + y));
+
+      expectObservable(result).toBe(expected);
+      expectSubscriptions(e1.subscriptions).toBe(e1subs);
+      expectSubscriptions(e2.subscriptions).toBe(e2subs);
+    });
   });
 
   it('should work with empty and empty', () => {
-    const e1 = cold('|');
-    const e1subs =  '(^!)';
-    const e2 = cold('|');
-    const e2subs =  '(^!)';
-    const expected = '|';
-
-    const result = of(e1, e2).pipe(combineAll((x, y) => x + y));
-
-    expectObservable(result).toBe(expected);
-    expectSubscriptions(e1.subscriptions).toBe(e1subs);
-    expectSubscriptions(e2.subscriptions).toBe(e2subs);
+    testScheduler.run(({ cold, expectObservable, expectSubscriptions }) => {
+      const e1 = cold('|');
+      const e1subs = ' (^!)';
+      const e2 = cold('|');
+      const e2subs = ' (^!)';
+      const expected = '|';
```

Comment:
I think `expected` was misaligned in the previous version. It should line up with the other bars.

Reply:
I'd think we would want to align on the ^ in`(^!)`, but it looks like we align on the opening parenthesis in other examples so I'll do that here - is that right?

Reply:
The `(^!)` is a subscription marble: `^` represents subscription and `!` represents unsubscription. The parentheses mean that the contents occur in the frame at which the `(` is placed, so, yeah, it needs to align with the `(`.

Note that `^` has a different meaning when it's used in an observable's marble diagram. There it represents the zero frame.